packages/suite/src/hooks/wallet/coinmarket/form/common/useCoinmarketBuildAccountGroups.ts

```diff
@@ -11,10 +12,28 @@ import {
 } from 'src/types/coinmarket/coinmarket';
 import { coinmarketBuildAccountOptions } from 'src/utils/wallet/coinmarket/coinmarketUtils';
 
+const filterNonEmptyAccounts = (account: Account) => {
+    return !account.empty || (account.tokens && account.tokens.length > 0);
+};
+
+const useValidAccounts = (type: CoinmarketTradeType) => {
+    const accounts = useSelector(selectAccounts);
+
+    return useMemo(() => {
+        switch (type) {
+            case 'sell':
+            case 'exchange':
+                return accounts.filter(filterNonEmptyAccounts);
+            case 'buy':
+                return accounts;
+        }
+    }, [type, accounts]);
```

Comment:
I prefer this sample of code, but it is 👌.

```suggestion
        if (type !== 'buy') return accounts.filter(filterNonEmptyAccounts);
        return accounts;
    }, [type, accounts]);
```

Reply:
done